pkg/catalog/catalog.go

```diff
@@ -62,15 +64,47 @@ func (c *Catalog) Get(sourceName, profileName string) *profilesv1.ProfileDescrip
 }
 
 // GetWithVersion returns the profile description `profileName` with the given version.
-func (c *Catalog) GetWithVersion(sourceName, profileName, version string) *profilesv1.ProfileDescription {
+func (c *Catalog) GetWithVersion(sourceName, profileName, profileVersion string) *profilesv1.ProfileDescription {
 	profiles, ok := c.m.Load(sourceName)
 	if !ok {
 		return nil
 	}
+
+	if profileVersion == "latest" {
+		return getLatestVersion(profiles.([]profilesv1.ProfileDescription), profileName)
+	}
+
 	for _, p := range profiles.([]profilesv1.ProfileDescription) {
-		if p.Name == profileName && p.CatalogSource == sourceName && p.Version == version {
+		if p.Name == profileName && p.CatalogSource == sourceName && p.Version == profileVersion {
 			return p.DeepCopy()
 		}
 	}
 	return nil
 }
+
+type profileDescriptionWithVersion struct {
+	profileDescription profilesv1.ProfileDescription
+	semverVersion      *version.Version
+}
+
+func getLatestVersion(profiles []profilesv1.ProfileDescription, profileName string) *profilesv1.ProfileDescription {
+	var profilesWithValidVersion []profileDescriptionWithVersion
+
+	for _, p := range profiles {
+		v, err := version.NewVersion(p.Version)
+		if err != nil {
+			continue
+		}
```

Comment:
I think we should validate a catalog entry has a valid semver version somewhere else, so here it just ignores the profile if its not valid

Reply:
where is the somewhere else?

Reply:
ideally a validation webhook, to give the error back to the user at `kubectl create` time

Reply:
cool 👍